python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -264,43 +354,99 @@ def _create_streaming_chat_message_content(
-                    contentItems.append(
-                        ImageContentItem(
-                            image_url=ImageUrl(url=item.data_uri or str(item.uri), detail=ImageDetailLevel.Auto)
-                        )
-                    )
-                else:
-                    logger.warning(
-                        "Unsupported item type in User message while formatting chat history for Azure AI"
-                        f" Inference: {type(item)}"
-                    )
-            chat_request_messages.append(_MESSAGE_CONVERTER[message.role](content=contentItems))
+            chat_request_messages.append(MESSAGE_CONVERTERS[message.role](message))
 
         return chat_request_messages
 
+    def _get_metadata_from_response(self, response: ChatCompletions | AsyncStreamingChatCompletions) -> dict[str, Any]:
+        """Get metadata from the response.
+
+        Args:
+            response: The response from the service.
+
+        Returns:
+            A dictionary containing metadata.
+        """
+        return {
+            "id": response.id,
+            "model": response.model,
+            "created": response.created,
+            "usage": response.usage,
+        }
+
+    def _verify_function_choice_behavior(
+        self,
+        settings: AzureAIInferenceChatPromptExecutionSettings,
+        kernel: Kernel,
+        arguments: KernelArguments,
+    ):
+        """Verify the function choice behavior."""
+        if settings.function_choice_behavior is not None:
+            if kernel is None:
+                raise ServiceInvalidExecutionSettingsError("Kernel is required for tool calls.")
+            if arguments is None and settings.function_choice_behavior.auto_invoke_kernel_functions:
+                raise ServiceInvalidExecutionSettingsError("Kernel arguments are required for auto tool calls.")
+            if settings.extra_parameters is not None and settings.extra_parameters.get("n", 1) > 1:
+                # Currently only OpenAI models allow multiple completions but the Azure AI Inference service
+                # does not expose the functionality directly. If users want to have more than 1 responses, they
+                # need to configure `extra_parameters` with a key of "n" and a value greater than 1.
+                raise ServiceInvalidExecutionSettingsError(
+                    "Auto invocation of tool calls may only be used with a single completion."
+                )
+
+    def _configure_function_choice_behavior(
+        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel
+    ):
+        """Configure the function choice behavior to include the kernel functions."""
+        settings.function_choice_behavior.configure(
+            kernel=kernel, update_settings_callback=update_settings_from_function_call_configuration, settings=settings
+        )
+
+    async def _process_function_calls(
```

Comment:
this has been marked deprecated in openai, moved the core functions into Kernel, should be done here as well!